### PR TITLE
{bio}[foss/2021b] methylpy v1.2.9 w/ Python 3.9.6

### PR DESCRIPTION
**Pull request by ItIsI-Orient**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by ItIsI-Orient** [comment]
Test report by @ItIsI-Orient
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3205.victini.os - Linux UNKNOWN UNKNOWN, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/4fc5c0fa3358d0af2fe1b2b9607ad8cb for a full test report.

**Comment by ItIsI-Orient** [comment]
Please, merge dependencies before testing:

- https://github.com/easybuilders/easybuild-easyconfigs/pull/14790
- https://github.com/easybuilders/easybuild-easyconfigs/pull/14791

File: easybuild/easyconfigs/m/methylpy/methylpy-1.2.9-foss-2021b-fixSetupConfig.patch [hunk 1]
```diff
@@ -0,0 +1,7 @@
+--- setup.cfg.orig	2022-01-19 19:34:38.630926000 +0100
++++ setup.cfg	2022-01-19 19:34:44.636856000 +0100
+@@ -1,2 +1,2 @@
+ [metadata]
+-description-file = README.md
++description_file = README.md
```
Comment (boegel) [inline easybuild/easyconfigs/m/methylpy/methylpy-1.2.9-foss-2021b-fixSetupConfig.patch:6]: For a simple change like this, why not use a `sed` command in `preinstallopts` instead, and remove the patch file?

```python
preinstallopts = "sed -i 's/description-file/description_file/g' setup.cfg && "
```

(the naming of the patch file could also be better, since the change is totally unrelated to the toolchain)

File: easybuild/easyconfigs/m/methylpy/methylpy-1.2.9-foss-2021b.eb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    ('Kent_tools', '422'),
+]
+
+use_pip = True
+download_dep_fail = True
+sanity_pip_check = True
+
+sanity_check_paths = {
+    'files': [],
+    'dirs': ['lib/python%(pyshortver)s/site-packages'],
```
Comment (boegel) [inline easybuild/easyconfigs/m/methylpy/methylpy-1.2.9-foss-2021b.eb:38]: This is the default for `PythonPackage`, so you can remove `sanity_check_paths`

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
@ItIsI-Orient The `tbb` issue pops up because we already have easyconfigs with a `2021b` toolchain that depend on `tbb` 2020.3, and we're introducing `tbb` 2021.4.0 as dependency for `methylpy` here via `Bowtie` (which was merged in #14791).

It probably makes sense to downgrade back to `tbb` 2020.3 in that `Bowtie` easyconfig, and I think it's fine to do that in this PR here (since that provides some context)...

**Comment by ItIsI-Orient** [comment]
Fixed version uploaded here, created PR to remove the version that is using incompatible tbb: https://github.com/easybuilders/easybuild-easyconfigs/pull/14891

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3164.skitty.os - Linux UNKNOWN UNKNOWN, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/a3be842610464e6d9139fbf493b0e767 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3112.skitty.os - Linux UNKNOWN UNKNOWN, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/50c885a14c9d405b51a7e06a806e724e for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @ItIsI-Orient!
